framework/wazuh/utils.py

```diff
@@ -613,7 +614,8 @@ class WazuhDBQuery(object):
     This class describes a database query for wazuh
     """
     def __init__(self, offset, limit, table, sort, search, select, query, fields, default_sort_field, db_path, count,
-                 get_data, default_sort_order='ASC', filters={}, min_select_fields=set(), date_fields=set(), extra_fields=set()):
+                 get_data, default_sort_order='ASC', filters={}, min_select_fields=set(), date_fields=set(),
+                 extra_fields=set(), backend='sqlite3', agent_id=None):
```

Comment:
The `backend` param could be redesign in a smarter way. Maybe we could isolate the lines in this code where you make a conditional based on the backend into separate methods of a new class `DBBackend`. Then you could build a `WazuhDBBackend` and a `Sqlite3Backend` classes. Each one of these classes would implement the specific logic to deal with different databases o services (like wazuh-db).

Reply:
Hello @crd1985,

I have added new commits including the suggested refactor. All unit tests are correct:
```python
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_ciscat.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 1 item

wazuh/tests/test_ciscat.py .                                                                                                                                                                                                                                          [100%]

========================================================================================================================= 1 passed in 0.25 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_syscollector.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 16 items

wazuh/tests/test_syscollector.py ................                                                                                                                                                                                                                     [100%]

========================================================================================================================= 16 passed in 0.36 seconds =========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_security_configuration_assessment.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 4 items

wazuh/tests/test_security_configuration_assessment.py ....                                                                                                                                                                                                            [100%]

========================================================================================================================= 4 passed in 0.27 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_rootcheck.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 9 items

wazuh/tests/test_rootcheck.py .........                                                                                                                                                                                                                               [100%]

========================================================================================================================= 9 passed in 0.31 seconds ==========================================================================================================================
# /var/ossec/framework/python/bin/pytest wazuh/tests/test_agent.py
============================================================================================================================ test session starts ============================================================================================================================
platform linux -- Python 3.7.2, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
rootdir: /home/vagrant/GitHub/wazuh/framework, inifile:
collected 39 items

wazuh/tests/test_agent.py .......................................                                                                                                                                                                                                     [100%]

========================================================================================================================= 39 passed in 0.45 seconds =========================================================================================================================
```